include/util/static_rtree.hpp

```diff
@@ -432,11 +418,6 @@ class StaticRTree
         std::priority_queue<QueryCandidate> traversal_queue;
         traversal_queue.push(QueryCandidate{0, TreeIndex{0}});
 
-        // we need to limit the size of this vector to MAX_NUM_CANDIDATES
```

Comment:
That is a good point. Actually if we make this all const, we could get around the copy-logic to create `PhantomNode` from this. Also we could slim down the `Hint` by just storing these indices. I'll ticket it.